src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -439,6 +439,20 @@ public void evictCache() {
         .thenAccept(aVoid -> noCatch(() -> addSkirmishMap(getPathForMap(folderName))));
   }
 
+  public CompletableFuture<List<MapBean>> getOwnedMaps(int playerId, int loadMoreCount, int page) {
+    return fafService.getOwnedMaps(playerId, loadMoreCount, page);
+  }
+
+  public CompletableFuture<Void> hideMapVersion(MapBean map) {
+    evictCache();
+    return fafService.hideMapVersion(map);
+  }
+
+  public CompletableFuture<Void> unrankMapVersion(MapBean map) {
+    evictCache();
```

Comment:
This won't evict the cache since the call isn't made through a spring bean proxy but directly within the class.
To make this work, the service needs to have itself injected by @Autowired and use that reference instead